application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml

```diff
@@ -59,13 +61,19 @@
       #set($list = $obj.getProperty('flash').getValue())
       #if($list == $util.getNull())
         #set($discard = $obj.set('flash', [$xcontext.user]))
-        #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #elseif($list.class.name == 'com.xpn.xwiki.objects.ListProperty$NotifyList' &amp;&amp; !$list.contains($xcontext.user))
         #set($discard = $list.add($xcontext.user))
         #set($discard = $obj.set('flash', $list))
-        #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #end
     #end
+    ## Check if the message should be marked as seen today.
+    #set ($msgObj = $msgdoc.getObject('Flash.FlashValidationClass'))
+    #set ($seenDate = $msgObj.getValue('seenDate'))
+    #set ($wasSeenToday = $datetool.format($dateFormat, $dateNow) == $datetool.format($dateFormat, $seenDate))
+    #if (!$seen || !$wasSeenToday)
+      #set ($discard = $msgObj.set('seenDate', $dateNow))
+      #set ($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
```

Comment:
This requires PR so it would be good to add a ``RequiredRights`` object to this page (that is the best practice).

Reply:
Would be nice to have a java component and a script service to deal with this save and thus to avoid the usage of `RequiredRights`. WDYT?

Reply:
It would probably require some refactoring because you cannot simply move the ``saveAsAuthor`` to a script service because anyone would be able to call it. I would to it in a separate issue.

Reply:
Would be cleaner indeed. I'll work on that on a different PR. Thanks!